integration/src/test/resources/org/mule/test/components/redelivery-policy-config.xml

```diff
@@ -95,4 +120,43 @@
             </on-error-propagate>
         </error-handler>
     </flow>
+
+    <flow name="redeliveryInvalidMessageIdWithTransactionalSourceAndCustomErrorHandlerDispatch">
+        <vm:publish-consume queueName="Q5" config-ref="VM_Config"/>
+    </flow>
+
+    <flow name="redeliveryInvalidMessageIdWithTransactionalSourceAndCustomErrorHandlerProcess">
+        <vm:listener queueName="Q5" config-ref="VM_Config" transactionalAction="ALWAYS_BEGIN">
+            <redelivery-policy idExpression="!null.a"/>
```

Comment:
is this of without the `#[]` marker?

Reply:
It actually works seamlessly with or without the `#[]` expression marker.